pilot/pkg/serviceregistry/serviceentry/controller.go

```diff
@@ -897,27 +891,66 @@ func autoAllocateIPs(services []*model.Service) []*model.Service {
 		// 3. the hostname is not a wildcard
 		if svc.DefaultAddress == constants.UnspecifiedIP && !svc.Hostname.IsWildCarded() &&
 			svc.Resolution != model.Passthrough {
-			n := svc.Hostname.String()
-			if v, ok := hnMap[n]; ok {
-				log.Debugf("Reuse IP for domain %s", n)
-
-				setAutoAllocatedIPs(svc, v)
+			hash.Write([]byte(svc.Attributes.Namespace + "/" + svc.Hostname.String()))
+			// First hash is calculated by
+			s := hash.Sum32()
+			fh := s % uint32(maxIPs)
```

Comment:
fh = firstHash? maybe need to be more clear

Reply:
yes